src/api/firebase.js

```diff
@@ -303,3 +303,42 @@ export async function deleteItem() {
 	 * this function must accept!
 	 */
 }
+
+export function comparePurchaseUrgency(data) {
+	const currentDate = new Date();
+
+	const updatedData = data.map((item) => {
+		// calculate time difference directly without using the absolute value built into getDaysBetweenDates function
+		const daysUntilNextPurchase = Math.floor(
+			(item.dateNextPurchased.toDate() - currentDate) / 86400000,
+		);
+
+		const daysSinceLastPurchase = item.dateLastPurchased
+			? getDaysBetweenDates(item.dateLastPurchased.toDate(), currentDate)
+			: null;
+
+		const isOverdue =
+			daysUntilNextPurchase < 0 &&
+			(daysSinceLastPurchase === null || daysSinceLastPurchase < 60);
+
+		return { ...item, daysUntilNextPurchase, daysSinceLastPurchase, isOverdue };
+	});
+
+	return updatedData.sort((itemA, itemB) => {
```

Comment:
Suggestion to improve readability by comparing within each category that itemA !== itemB and and subtracting one from the other which provides a comparison implicitly through the .sort() method.

Reply:
Thanks for the suggestion! Is this what you were thinking? For example for prioritizing active items:

if (itemA.daysSinceLastPurchase !== itemB.daysSinceLastPurchase) return itemA - itemB